### PR TITLE
Fixed issue #603

### PR DESCRIPTION
**Pull request by fabienrenaud**
`_currInputProcessed`, `_currInputRowStart`, and `_nameStartOffset` must be updated before `_inputPtr`

See #603

### CODE REVIEW

File: src/test/java/com/fasterxml/jackson/core/read/LocationOffsetsTest.java [hunk 1]
```diff
@@ -3,6 +3,11 @@
 import com.fasterxml.jackson.core.*;
 import com.fasterxml.jackson.core.json.JsonFactory;
 
+import java.io.IOException;
+import java.util.Base64;
```
Comment (cowtowncoder) [inline src/test/java/com/fasterxml/jackson/core/read/LocationOffsetsTest.java]: Can not use until 3.0 (part of JDK 8). Same for a few other things, will make necessary changes.

**Comment by cowtowncoder** [comment]
Thank you for the fix(es)!

Due to minor JDK 8 dependencies here I think I'll merge changes manually, starting with 2.10 branch. But keeping essential logic.

**Comment by fabienrenaud** [comment]
@cowtowncoder I can replace the JDK8 dependencies with this function if you'd like:
```
    private String generateRandomAlpha(int length)
    {
        StringBuilder sb = new StringBuilder(length);
        Random rnd = new Random(length);
        for (int i = 0; i < length; ++i) {
            // let's limit it not to include surrogate pairs:
            char ch = (char) ('A' + rnd.nextInt(26));
            sb.append(ch);
        }
        return sb.toString();
    }
```

File: src/test/java/com/fasterxml/jackson/core/read/LocationOffsetsTest.java (hunk 1 above)
Comment (fabienrenaud) [reply src/test/java/com/fasterxml/jackson/core/read/LocationOffsetsTest.java]: I just replaced it with a `generateRandomAlpha` method.

**Comment by cowtowncoder** [comment]
Jackson has full Base64 functionality in itself ....

**Comment by fabienrenaud** [comment]
I don't really need base64 for this test, so it's okay...

**Pull request merged**